lib/Doctrine/Persistence/ObjectRepository.php

```diff
@@ -23,9 +23,11 @@ public function find($id) : ?object;
     /**
      * Finds all objects in the repository.
      *
+     * @param array<string, string> $orderBy
+     *
      * @return array<int, object> The objects.
      */
-    public function findAll() : array;
+    public function findAll(?array $orderBy) : array;
```

Comment:
It shouldn't be non-optional nullable array, but array with empty array as default. Anyway I'm not sure we will accept this BC break, but I'm pro.

Reply:
nullable array seems to be in the signature of `findBy` which is what `findAll` calls in `EntityRepository`.

```php
public function findBy(array $criteria, ?array $orderBy = null, $limit = null, $offset = null)
{
    $persister = $this->em->getUnitOfWork()->getEntityPersister($this->entityName);

    return $persister->loadAll($criteria, $orderBy ?? [], $limit, $offset);
}
```
it also seems that it will default to an empty array regardless.

Anyways, happy to change, just that using the default of `findBy` seemed the sensible thing to do.

Also, unless I'm missing something why would this be a BC? Existing code unaware of this argument will continue to work as before, no?

Reply:
Regarding the BC stuff, I just realized there wasn't a default value...

Reply:
Null default was oversight, please use empty array https://github.com/doctrine/orm/blob/0c2ccec9151fd0b44a60f9cc149e2a107d380799/lib/Doctrine/ORM/EntityRepository.php#L126

But BC break is not just because of that https://3v4l.org/2OnNR

Reply:
that's a BC break for implementers, not for consumers. But that's still a BC break.